slsDetectorGui/src/qDetectorMain.cpp

```diff
@@ -116,10 +116,10 @@ qDetectorMain::qDetectorMain(int argc, char **argv, QApplication *app,
 }
 
 qDetectorMain::~qDetectorMain() {
-    if (myDet)
-        delete myDet;
-    if (myPlot)
-        delete myPlot;
+    if (det)
```

Comment:
You don't have to check for null pointer before deleting the object. If it is a null pointer the
delete does nothing. Only when reading or writing to a pointer you need to check.

Reply:
done